### PR TITLE
Added both reporting articles

### PR DESCRIPTION
**Pull request by maudnals**
Fixes #SOME_ISSUE_NUMBER

Changes proposed in this pull request:

- 
- 
-

### CODE REVIEW

File: src/site/content/en/blog/reporting-api-migration/index.md [hunk 1]
```diff
@@ -0,0 +1,471 @@
+---
+layout: post
+title: Migrate to the Reporting API v1
```
Comment (mihajlija) [inline src/site/content/en/blog/reporting-api-migration/index.md:3]:
```suggestion
title: Migrate to Reporting API v1
```

File: src/site/content/en/blog/reporting-api-migration/index.md [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+subhead: |
+  A new version of the Reporting API is available. It's simpler, more private, and more likely to be supported across browsers.
+authors:
+  - maudn
+date: 2021-06-14
+updated: 2021-06-14
+description: |
+  Migrate to the new Reporting API.
```
Comment (mihajlija) [inline src/site/content/en/blog/reporting-api-migration/index.md:11]:
```suggestion
  A new version of the Reporting API is available. It's simpler, more private, and more likely to be supported across browsers. This post goes over how to migrate to the new version of the Reporting API.
```

File: src/site/content/en/blog/reporting-api-migration/index.md [hunk 3]
```diff
@@ -0,0 +1,471 @@
...
+
+If you're adding reporting functionality to your site just now, use only v1 (`Reporting-Endpoints` header).
+
+Make sure to set the `Reporting-Endpoints` header on all responses that might generate
+reports.
+
+If you're maintaining an endpoint service or are operating your own, expect **more
+traffic** as you or external developers migrate to v1 (`Reporting-Endpoints` header).
+
+{% endBanner %}
```
Comment (mihajlija) [inline src/site/content/en/blog/reporting-api-migration/index.md:50]:
```suggestion

If you already have reporting functionality for your site, **migrate to v1**
(`Reporting-Endpoints` header) but keep the v0 header (`Report-To`) around for some time.

If you're adding reporting functionality to your site just now, use only v1 (`Reporting-Endpoints` header).

Make sure to set the `Reporting-Endpoints` header on all responses that might generate
reports.

If you're maintaining an endpoint service or are operating your own, expect **more
traffic** as you or external developers migrate to v1 (`Reporting-Endpoints` header).

```

Comment (mihajlija) [reply src/site/content/en/blog/reporting-api-migration/index.md:50]: This is too much text for a banner, I think it's easier to read on a plain background. It's already in its own section that should be enough.

File: src/site/content/en/blog/reporting-api-migration/index.md [hunk 4]
```diff
@@ -0,0 +1,471 @@
...
+If you use [Network Error Logging](https://w3c.github.io/network-error-logging/), continue using `Report-To` (v0) because Network Error Logging isn't
+supported in the Reporting API v1. A new mechanism for Network Error Logging will be
+developed⏤until that's available, the Reporting API v0 should be used.
+**As of June 2021**, this new mechanism is not shipped yet.
+
+{% endBanner %}
+
+## Demo and code
+
+- [Demo site: new reporting API (v1)](https://reporting-api-demo.glitch.me)
```
Comment (mihajlija) [inline src/site/content/en/blog/reporting-api-migration/index.md:67]: I'm on Chrome 91.0.4472.77 and at first I got 0 reports on the reporting endpoint page. Tried refreshing multiple times. I

When I tried it in Canary, I initially got 1 report, then it jumped to 4 reports. I had that same page opened in my Chrome 91 instance and it also got updated to show 4 reports.

I also couldn't find the --short-reporting-delay flag in either 91 or Canary.

AI:
- Verify that this really works in Chrome 91.
- Check the demo behaviour when multiple users visit it.
- Clarify where to find the flag and what it does.

Another thing: You might want to "boost" this app on Glitch, it was asleep when I clicked and it took a while for it to load.

Comment (mihajlija) [reply src/site/content/en/blog/reporting-api-migration/index.md:67]: Once that's cleared up, consider embedding this demo. I know there's not much to see on that first demo page, but at least readers will get a preview of what the demo is about.

Comment (maudnals) [reply src/site/content/en/blog/reporting-api-migration/index.md:65]: > When I tried it in Canary, I initially got 1 report, then it jumped to 4 reports
I think this is due to the missing flag --short-reporting-delay.

>  I also couldn't find the --short-reporting-delay flag in either 91 or Canary.
Oh so it's a CLI flag only, not available from the UI! Will need to update the text so readers aren't confused.

Can you maybe try the demo with the flag on (via your terminal)? Instructions are in the article.
Thank you!

File: src/site/content/en/blog/reporting-api-migration/index.md [hunk 5]
```diff
@@ -0,0 +1,469 @@
...
+
+- The format and structure of the reports is unchanged.
+- The request sent by the browser to the endpoint remains a POST request of "Content-type"
+  `application/reports+json`.
+- Mapping certain endpoints to certain report types is supported in both v0 and v1.
+- The formatting of the rules that generate reports when violated is unchanged.
+- The role of the default endpoint is unchanged.
+- The Reporting API v1 has no impact on the [ReportingObserver](/reporting-observer).
+  ReportingObserver continues getting access to all observable reports, and their format is
+  identical.
```
Comment (mihajlija) [inline src/site/content/en/blog/reporting-api-migration/index.md:127]:
```suggestion
- The Reporting API v1 has no impact on the [`ReportingObserver`](/reporting-observer).
  `ReportingObserver` continues getting access to all observable reports, and their format is
  identical.
```

File: src/site/content/en/blog/reporting-api-migration/index.md [hunk 6]
```diff
@@ -0,0 +1,469 @@
...
+   responses that might generate reports.
+
+   With v0, you could set reporting endpoints on some responses only, and other documents
+   (pages) on that origin would use this "ambient" endpoint. With v1, because of the
+   difference in scoping, you need to set the `Reporting-Endpoints` header on all
+   responses that might generate reports.
+
+3. **Step 3 (start later):** Once all or most of your users have updated to later Chrome
+   or Edge installs (93 and later), remove **Report-To** (v0) and keep only
+   `Reporting-Endpoints`.
```
Comment (mihajlija) [inline src/site/content/en/blog/reporting-api-migration/index.md:278]:
```suggestion
   or Edge installs (93 and later), remove `Report-To` (v0) and keep only
   `Reporting-Endpoints`.
```

File: src/site/content/en/blog/reporting-api-migration/index.md [hunk 7]
```diff
@@ -0,0 +1,469 @@
...
+
+   One exception: if you do need Network Error Logging reports, keep **Report-To** until a
+   new mechanim is in place for Network Error Logging.
```
Comment (mihajlija) [inline src/site/content/en/blog/reporting-api-migration/index.md:281]:
```suggestion
   One exception: if you do need Network Error Logging reports, keep `Report-To` until a
   new mechanism is in place for Network Error Logging.
```

File: src/site/content/en/blog/reporting-api-migration/index.md [hunk 8]
```diff
@@ -0,0 +1,469 @@
...
+1. **Step 1 (do now)**: If you haven't added it yet, add `report-to` alongside `report-uri`.
+   Browsers that support only `report-uri` (Firefox) will use `report-uri`, and browsers
+   that also support `report-to`(Chrome, Edge) will use `report-to`.
+
+2. **Step 2 (do now)**: To specify the named endpoints you'll use in `report-to`, use both
+   headers `Report-To` and `Reporting-Endpoints`. This ensures that you get reports from
+   both older and newer Chrome and Edge clients.
+
+3. **Step 3 (start later):** Once all or most of your users have updated to later Chrome
+   or Edge installs (93 and later), remove **Report-To** (v0) and keep only
```
Comment (mihajlija) [inline src/site/content/en/blog/reporting-api-migration/index.md:313]:
```suggestion
   or Edge installs (93 and later), remove `Report-To` (v0) and keep only
```

File: src/site/content/en/blog/reporting-api-migration/index.md [hunk 9]
```diff
@@ -0,0 +1,469 @@
...
+   `Reporting-Endpoints`. Keep `report-uri` so you still get reports for browsers that
+   only support it.
+
+See code examples in the [migration cookbook](#csp-reporting-migration).
+
+### Migration Cookbook
```
Comment (mihajlija) [inline src/site/content/en/blog/reporting-api-migration/index.md:319]:
```suggestion
### Migration cookbook
```

File: src/site/content/en/blog/reporting-api-migration/index.md [hunk 10]
```diff
@@ -0,0 +1,469 @@
...
+```http
+Reporting-Endpoints: main-endpoint="https://reports.example/main", default="https://reports.example/default"
+Report-To: { group: "main-endpoint", "max_age": 86400, "endpoints": [ { "url": "https://reports.example/main" }] }, { group: "default-endpoint", "max_age": 86400, "endpoints": [ { "url": "https://reports.example/default" }] }
+```
+
+{% CompareCaption %}
+
+If you already have reporting functionality in your site, keep `Report-To` **only temporarily** (until most Chrome and Edge clients have been updated) to avoid losing reports.
+
+If you need Network Error Logging, keep `Report-To` **until Network Error Logging has a new replacement**.
```
Comment (mihajlija) [inline src/site/content/en/blog/reporting-api-migration/index.md:342]:
```suggestion
If you need Network Error Logging, keep `Report-To` **until Network Error Logging replacement is available**.
```

File: src/site/content/en/blog/reporting-api/index.md [hunk 11]
```diff
@@ -0,0 +1,660 @@
...
+Some errors only occur in production. You never see them locally or during development
+because **real users**, **real networks**, and **real devices** change the game. The
+Reporting API helps catch some of these errors across your site and transmit them to an
+endpoint you've specified.
+
+This API enables you to activate a reporting mechanism that's **operated by the browser**.
+When the Reporting API configured on your site, **reports can be generated** when certain
+policies are violated (or when deprecated APIs are used, when the browser intervenes on
+your site or crashes, and more). The browser then sends those report(s) to the
+**endpoint(s) i.e. report collectors** you've configured.
```
Comment (mihajlija) [inline src/site/content/en/blog/reporting-api/index.md:36]:
```suggestion
When the Reporting API is configured on your site, **reports can be generated** when certain
policies are violated (or when deprecated APIs are used, when the browser intervenes on
your site or crashes, and more). The browser then sends those report(s) to the
**endpoint(s)**, that is report collectors, you've configured.
```

File: src/site/content/en/blog/reporting-api/index.md [hunk 12]
```diff
@@ -0,0 +1,660 @@
...
+
+💥 [Demo site: report generation page](https://reporting-api-demo.glitch.me). This page
+uses the new Reporting API via the `Reporting-Endpoints` header. See the
+[code](https://glitch.com/edit/#!/reporting-api-demo).
+
+📬 [Demo site: reporting endpoint](https://reports-endpoint.glitch.me). This server
+receives and displays reports. See the
+[code](https://glitch.com/edit/#!/reports-endpoint).
+
+{% endBanner %}
```
Comment (mihajlija) [inline src/site/content/en/blog/reporting-api/index.md:62]:
```suggestion

💥 [Demo site: report generation page](https://reporting-api-demo.glitch.me). This page
uses the new Reporting API via the `Reporting-Endpoints` header. See the
[code](https://glitch.com/edit/#!/reporting-api-demo).

📬 [Demo site: reporting endpoint](https://reports-endpoint.glitch.me). This server
receives and displays reports. See the
[code](https://glitch.com/edit/#!/reports-endpoint).

```

Comment (mihajlija) [reply src/site/content/en/blog/reporting-api/index.md:62]: Removing the banner for readability here too. I'm torn about the emoji, they are whimsical but for screen readers it's just clutter. It reads as "Explosion demo site..." and more jarringly, "Open letterbox with raised flag demo site..."

File: src/site/content/en/blog/reporting-api/index.md [hunk 13]
```diff
@@ -0,0 +1,660 @@
...
+        <td>Your site has a document policy that prevents usage of <code>document.write</code>, but a script on your page tries to use <code>document.write</code>.</td>
+      </tr>
+      <tr>
+        <td>Deprecation warning</td>
+        <td>Your site is using an API that is deprecated, directly or via a top-level third-party script.
+        </td>
+      </tr>
+      <tr>
+        <td>Intervention</td>
+        <td>Your site is trying to do something that the browser decides not to honor, for security, performance or user experience reasons. Example in Chrome: your site uses <a href="https://developers.google.com/web/updates/2016/08/removing-document-write"><code>document.write</code> on slow networks</a> or uses <code>navigator.vibrate</code> in a cross-origin frame that the user hasn't interacted with yet.</td>
```
Comment (mihajlija) [inline src/site/content/en/blog/reporting-api/index.md:105]:
```suggestion
<div class="w-table-wrapper">
  <table>
    <thead>
      <tr>
        <th>Report type</th>
        <th>Example of a situation where a report would be generated</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td><a href="https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP">CSP</a> violation (Level 3 only)</td>
        <td>You've set a <code>Content-Security-Policy</code> on your site, but one of your pages is trying to load a script that's not allowed by your CSP.</td>
      </tr>
      <tr>
        <td><a href="/why-coop-coep/#coop">COOP</a> violation</td>
        <td>You've set a <code>Cross-Origin-Opener-Policy</code> on your site, but a cross-origin window is trying to interact directly with your document.</td>
      </tr>
      <tr>
        <td><a href="/why-coop-coep/#coep">COEP</a> violation</td>
        <td>You've set a <code>Cross-Origin-Embedder-Policy</code> on your site, but a page on your site includes a cross-origin iframe that has not opted into being loaded by cross-origin documents.</td>
      </tr>
      <tr>
        <td>Document Policy violation</td>
        <td>Your site has a document policy that prevents usage of <code>document.write</code>, but a script on your page tries to use <code>document.write</code>.</td>
      </tr>
      <tr>
        <td>Deprecation warning</td>
        <td>Your site is using an API that is deprecated, directly or via a top-level third-party script.
        </td>
      </tr>
      <tr>
        <td>Intervention</td>
        <td>Your site is trying to do something that the browser decides not to honor, for security, performance or user experience reasons. Example in Chrome: your site uses <a href="https://developers.google.com/web/updates/2016/08/removing-document-write"><code>document.write</code> on slow networks</a> or uses <code>navigator.vibrate</code> in a cross-origin frame that the user hasn't interacted with yet.</td>
```

File: src/site/content/en/blog/reporting-api/index.md [hunk 14]
```diff
@@ -0,0 +1,660 @@
...
+  </table>
+</div>
+
+{% Aside %}
+**[Permissions
+policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Feature-Policy)
+(formerly feature policy)** violation reports may be supported by default in the future.
+Right now, they're experimental. One example situation where such reports would be
+generated: if your site has a permission policy to prevent microphone usage, and a
+script is requesting audio input.
```
Comment (mihajlija) [inline src/site/content/en/blog/reporting-api/index.md:121]:
```suggestion
Right now, they're experimental. One example situation where such reports would be
generated is if your site has a permission policy to prevent microphone usage, and a
script is requesting audio input.
```

File: src/site/content/en/blog/reporting-api/index.md [hunk 15]
```diff
@@ -0,0 +1,660 @@
...
+
+{% endAside %}
+
+## How it works
+
+<figure class="w-figure">
+{% Img src="image/O2RNUyVSLubjvENAT3e7JSdqSOx1/RgiT3KRy6w99wG1cAhha.png", alt="Diagram summarizing the steps below, from report generation to report access by the developer", width="800", height="240" %}
+  <figcaption class="w-figcaption">
+    Diagram: how reports are generated and sent
```
Comment (mihajlija) [inline src/site/content/en/blog/reporting-api/index.md:130]:
```suggestion
    How reports are generated and sent.
```

File: src/site/content/en/blog/reporting-api/index.md [hunk 16]
```diff
@@ -0,0 +1,660 @@
...
+  guide](/reporting-api-migration).
+- In the new Reporting API, reports are scoped to **documents**. This means that for one
+  given origin, different documents (`site.example/page1` and `site.example/page2`) can
+  send reports to different endpoints. If you're migrating from the legacy Reporting API
+  to the new Reporting API, check the [migration guide](/reporting-api-migration).
+- Reports are sent with a delay to increase the chances that reports are sent in batches.
+  This saves bandwidth to be respectful to the user's network connection. This is
+  especially important on mobile.
+
+{% endAside %}
```
Comment (mihajlija) [inline src/site/content/en/blog/reporting-api/index.md:327]:
```suggestion
{% Aside %} 
- In the new Reporting API (v1), thee browser only batches together reports in the same
  request if they are generated by the same document (page). Reports are not sent in the
  same request if they come from different origins, or even different documents on the
  same origin such as different pages on the same website. Learn more in the [migration
  guide](/reporting-api-migration).
- In the new Reporting API, reports are scoped to **documents**. This means that for one
  given origin, different documents (`site.example/page1` and `site.example/page2`) can
  send reports to different endpoints. If you're migrating from the legacy Reporting API
  to the new Reporting API, check the [migration guide](/reporting-api-migration).
- Reports are sent with a delay to increase the chances that reports are sent in batches.
  This saves bandwidth to be respectful to the user's network connection. This is
  especially important on mobile.

{% endAside %}
```

File: src/site/content/en/blog/reporting-api/index.md [hunk 17]
```diff
@@ -0,0 +1,660 @@
...
+
+## Reports you get
+
+Let's assume that you've set up `Reporting-Endpoints` and a few policies on your page.
+
+✅ Reports that are generated due to violations of these policies (or deprecations) on your
+page will be sent to the endpoint(s) you've configured. This includes violations committed
+by **third-party scripts** running on your page.
+
+❌ Violations committed by a **cross-origin iframe** on your page will **not** be reported
```
Comment (mihajlija) [inline src/site/content/en/blog/reporting-api/index.md:337]: Again, emoji weirdness for screen readers. It reads as "Cross mark violations committed by a cross origin iframe" which could be confusing. I think it's best to not use emoji in this way in our technical docs and limit the use to what makes sense "semantically" rather than just visually. (Not saying you shouldn't use emoji at all!)

File: src/site/content/en/blog/reporting-api/index.md [hunk 18]
```diff
@@ -0,0 +1,660 @@
...
+to your endpoint(s), to ensure privacy. The iframe could set up its own reporting and even
+report to your site's (i.e. the first-party's) reporting service, but that's a decision
```
Comment (mihajlija) [inline src/site/content/en/blog/reporting-api/index.md:339]:
```suggestion
report to your site's (that is, the first-party's) reporting service, but that's a decision
```

File: src/site/content/en/blog/reporting-api/index.md [hunk 19]
```diff
@@ -0,0 +1,660 @@
...
+that the framed site would have to make.
+
+{% Aside 'gotchas' %}
+
+In Chrome DevTools, you'll see a console error or warning pop up for violations that are
```
Comment (mihajlija) [inline src/site/content/en/blog/reporting-api/index.md:344]:
```suggestion
In Chrome DevTools, you'll see a console error or warning pop-up for violations that are
```

File: src/site/content/en/blog/reporting-api/index.md [hunk 20]
```diff
@@ -0,0 +1,660 @@
...
+
+## Browser support
+
+The table below sums up browser support for the **Reporting API v1**, that is with the
+`Reporting-Endpoints` header. Browser support for the legacy Reporting API (`Report-To`
+header) is the same, except for one report type: Network Error Logging isn't supported in
+the new Reporting API. Read the [migration
+guide](/reporting-api-migration/#network-error-logging) for details.
+
+<div class="w-table-wrapper">
```
Comment (mihajlija) [inline src/site/content/en/blog/reporting-api/index.md:357]: I'll come back to this table formatting after I talk to Rob. Ideally, it would use only Yes/No in text and highlight the cells in green/red.

File: src/site/content/en/blog/reporting-api/index.md [hunk 21]
```diff
@@ -0,0 +1,660 @@
...
+specificities with `report-uri`. See details in the [migration
+guide](/reporting-api-migration).
+
+{% Aside %} One example where `...-Report-Only` headers are useful: let's assume you have
+a CSP set up for your site. You plan on tightening up your site's security by making your
+CSP stricter. Instead of directly deploying this new, stricter CSP, you first deploy it in
+`Report-Only` mode. This helps you monitor what your CSP _could have broken_ and tweak it,
+instead of actually inducing breakage for your end-users. {% endAside %}
+
+### ReportingObserver
```
Comment (mihajlija) [inline src/site/content/en/blog/reporting-api/index.md:420]:
```suggestion
### `ReportingObserver`
```

File: src/site/content/en/blog/reporting-api/index.md [hunk 22]
```diff
@@ -0,0 +1,660 @@
...
+
+You may want to monitor your site but have no possibility to set headers on your server.
+In that case, use [ReportingObserver](/reporting-observer).
```
Comment (mihajlija) [inline src/site/content/en/blog/reporting-api/index.md:423]:
```suggestion
In that case, use [`ReportingObserver`](/reporting-observer).
```

File: src/site/content/en/blog/reporting-api/index.md [hunk 23]
```diff
@@ -0,0 +1,660 @@
...
+
+This JavaScript API can observe simple client-side warnings like deprecation and
+intervention. It does not support other types of errors, like CSP or COOP/COEP violations.
+
+This API is also useful in case you need to react to violations in real-time.
+
+## How-to: Configure report sending
+
+### Configure the Reporting-Endpoints header
```
Comment (mihajlija) [inline src/site/content/en/blog/reporting-api/index.md:432]:
```suggestion
### Configure the `Reporting-Endpoints` header
```

File: src/site/content/en/blog/reporting-api/index.md [hunk 24]
```diff
@@ -0,0 +1,660 @@
...
+If you're migrating from the legacy Reporting API to the new Reporting API, it may make
+sense to configure **both** `Reporting-Endpoints` and `Report-To`. See details in the
+[migration guide](/reporting-api-migration/#migration-steps).
+
+```http
+Reporting-Endpoints: main-endpoint="https://reports.example/main", default="https://reports.example/default"
+Report-To: ...
+```
+
+**Note:** If you're using reporting for **Content-Security-Policy** violations via the
```
Comment (mihajlija) [inline src/site/content/en/blog/reporting-api/index.md:447]:
```suggestion
Note that if you're using reporting for `Content-Security-Policy` violations via the
```

File: src/site/content/en/blog/reporting-api/index.md [hunk 25]
```diff
@@ -0,0 +1,660 @@
...
+</div>
+
+## How-to: Receive reports
+
+When deciding where to send your reports, you have two options:
+
+- Option 1: Use an existing report collector service.
+- Option 2: Build and operate your own reporting collector.
+
+**Also note that can set multiple endpoints**. This can be handy to separate reports:
```
Comment (mihajlija) [inline src/site/content/en/blog/reporting-api/index.md:501]:
```suggestion
**Also note that you can set multiple endpoints**. This can be handy to separate reports:
```

File: src/site/content/en/blog/reporting-api/index.md [hunk 26]
```diff
@@ -0,0 +1,660 @@
...
+  `endpoint.example/coop` `endpoint.example/csp` `endpoint.example/default`
+
+### Option 1: Use an existing report collector service
+
+Some examples of report collector services are [report-uri](https://report-uri.com) and
+[uriports](https://uriports.com). If you know of other solutions, please [open an
+issue](https://github.com/GoogleChrome/web.dev/issues) to let us know, and we'll update
+this post!
+
+**🧐 Things to consider when selecting a report collector as a service:**
```
Comment (mihajlija) [inline src/site/content/en/blog/reporting-api/index.md:515]:
```suggestion
**Things to consider when selecting a report collector as a service** 🧐 
```
Works better this way when emoji is not before text. I can allow this! 🤓

File: src/site/content/en/blog/reporting-api/index.md [hunk 27]
```diff
@@ -0,0 +1,660 @@
...
+### Option 2: Build your own report collector
+
+Building your own server that receives reports isn't that trivial. To get started, you can
+use our lightweight boilerplate. It's built with express and can receive and display
+reports.
+
+- Head over to the [boilerplate report
+  collector](https://glitch.com/edit/#!/reports-endpoint)
+- Click **Remix to edit**
+- You now have your clone ✨ You can customize it for your own purposes.
```
Comment (mihajlija) [inline src/site/content/en/blog/reporting-api/index.md:534]:
```suggestion
1. Head over to the [boilerplate report
  collector](https://glitch.com/edit/#!/reports-endpoint).
- {% Instruction 'remix' %}
3. You now have your clone! You can customize it for your own purposes.
```

Comment (mihajlija) [reply src/site/content/en/blog/reporting-api/index.md:534]: Tip: Use [instructions components](https://web.dev/handbook/web-dev-components/#instructions) for Glitch instructions (helps with doc maintenance when Glitch changes UI). Also, use numbered lists for steps.

File: src/site/content/en/blog/reporting-api/index.md [hunk 28]
```diff
@@ -0,0 +1,660 @@
...
+
+**🧐 Things to consider when building your own report collector:**
```
Comment (mihajlija) [inline src/site/content/en/blog/reporting-api/index.md:536]:
```suggestion
**Things to consider when building your own report collector** 🧐 
```

File: src/site/content/en/blog/reporting-api/index.md [hunk 29]
```diff
@@ -0,0 +1,660 @@
...
+
+- If your endpoint URL lives on a different origin than your site, the endpoint should
+  support [CORS preflight
+  requests](https://developer.mozilla.org/en-US/docs/Glossary/Preflight_request).
+- To recognize reports requests sent by the browser to your endpoint, check for `POST`
+  requests with a `Content-Type` of `application/reports+json`.
+
+Our boilerplate report collector linked above does both these things.
```
Comment (mihajlija) [inline src/site/content/en/blog/reporting-api/index.md:544]:
```suggestion
Our boilerplate report collector linked above does both of these things.
```

File: src/site/content/en/blog/reporting-api/index.md [hunk 30]
```diff
@@ -0,0 +1,660 @@
...
+### Generating reports
+
+When debugging, you may want to intentionally violate your policies in order to trigger
+report sending. To see example code that violates policies and does other bad things that
+will trigger reports of all types, check out the [demo](#demo-and-code).
+
+### Saving time
+
+Reports may be sent with a delay, up to a minute, to preserve user experience. But when
+debugging, a minute is a long time 😴. Luckily, when debugging in Chrome, you can launch
```
Comment (mihajlija) [inline src/site/content/en/blog/reporting-api/index.md:557]:
```suggestion
debugging, a minute is a long time. 😴 Luckily, when debugging in Chrome, you can launch
```

File: src/site/content/en/blog/reporting-api/index.md [hunk 31]
```diff
@@ -0,0 +1,660 @@
...
+it with `--short-reporting-delay` to receive reports almost immediately. Run this command
+in your terminal:
+
+```bash
+path/to/executable/Chrome --short-reporting-delay
```
Comment (mihajlija) [inline src/site/content/en/blog/reporting-api/index.md:562]:
```suggestion
YOUR_PATH/TO/EXECUTABLE/Chrome --short-reporting-delay
```

Comment (mihajlija) [reply src/site/content/en/blog/reporting-api/index.md:562]: Trying to highlight that this is a placeholder path

**Review by mihajlija — CHANGES_REQUESTED** [review]
Almost LGTM, just want to make sure the demo requirements are checked and explained to readers correctly.

File: src/site/content/en/blog/reporting-api/index.md (hunk 21 above)
Comment (maudnals) [reply src/site/content/en/blog/reporting-api/index.md:418]: Cool TIL, I thought `` couldn't be used in headings.

File: src/site/content/en/blog/reporting-api-migration/index.md (hunk 4 above)
Comment (maudnals) [reply src/site/content/en/blog/reporting-api-migration/index.md:65]: The correct version number was 93. Done:
- Updated version number to 93 in the demo
- Clarified that the flags are CLI-only i.e. not in the Chrome UI, in both the demo page and here in the article where relevant https://deploy-preview-5580--web-dev-staging.netlify.app/reporting-api/#saving-time

> Another thing: You might want to "boost" this app on Glitch, it was asleep when I clicked and it took a while for it to load.
oh yes great tip. I thought that glitch demos under the web.dev team were automatically boosted. Is it not the cae and so I need to boost it manually?

File: src/site/content/en/blog/reporting-api/index.md (hunk 20 above)
Comment (maudnals) [reply src/site/content/en/blog/reporting-api/index.md:354]: SG! LMK when you have an update on this.
Side note: one thing I was looking at when preparing this table was [this example in the handbook](https://web.dev/handbook/web-dev-components/#:~:text=if%20you%20want%20content%20in) with the crosses and checks. IIUC it just happens to be a yes/no table, the reason it's in the handbook is to illustrate something else (alignment).

image

If yes/no tables are common, an example in the handbook on how to do these right would be super helpful!

File: src/site/content/en/blog/reporting-api/index.md (hunk 17 above)
Comment (maudnals) [reply src/site/content/en/blog/reporting-api/index.md:334]: SG, removed these emojis. Thanks for checking!

File: src/site/content/en/blog/reporting-api/index.md [hunk 32]
```diff
@@ -0,0 +1,662 @@
...
+### Option 2: Build your own report collector
+
+Building your own server that receives reports isn't that trivial. To get started, you can
+use our lightweight boilerplate. It's built with express and can receive and display
+reports.
+
+1. Head over to the [boilerplate report
+   collector](https://glitch.com/edit/#!/reports-endpoint).
+
+- {% Instruction 'remix' %}
```
Comment (maudnals) [inline src/site/content/en/blog/reporting-api/index.md]: @mihajlija Great tip!

File: src/site/content/en/blog/reporting-api/index.md (hunk 20 above)
Comment (mihajlija) [reply src/site/content/en/blog/reporting-api/index.md:353]: Thanks for bringing this to my attention, I'm gonna need to do a through review of use of tables on the site. 👷🏻‍♀️

**Comment by wolfib** [comment]
LGTM from DevTools perspective.

Only found a few typos in src/site/content/en/blog/reporting-api/index.md

L310: "thee"
L340: "pop up" (no hyphen imo because it's a verb)
L341:  "...committed by third-party scriptS ..."

**Comment by maudnals** [comment]
@wolfib Fixed! Thank you for your review.

File: src/site/content/en/blog/reporting-api-migration/index.md (hunk 4 above)
Comment (maudnals) [reply src/site/content/en/blog/reporting-api-migration/index.md:65]: Demos are now boosted.
(Note: TIL glitch demos under the web.dev team are automatically rate-UNlimited, but not automatically boosted. I purchased a Pro membership in order to boost them.)

File: src/site/content/en/blog/reporting-api/index.md [hunk 33]
```diff
@@ -0,0 +1,664 @@
...
+    <thead>
+      <tr>
+        <th>Report type</th>
+        <th>Example of a situation where a report would be generated</th>
+      </tr>
+    </thead>
+    <tbody>
+      <tr>
+        <td><a href="https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP">CSP</a> violation (Level 3 only)</td>
+        <td>You've set a <code>Content-Security-Policy</code> on your site, but one of your pages is trying to load a script that's not allowed by your CSP.</td>
```
Comment (agektmr) [inline src/site/content/en/blog/reporting-api/index.md:80]: I think a CSP is set against a page rather than a site, so `one of your pages` is somewhat confusing statement. I'd rephrase it to `You've set a <code>Content-Security-Policy</code> on one of your pages, but the page is trying to load a script that's not allowed by your CSP.`

File: src/site/content/en/blog/reporting-api/index.md [hunk 34]
```diff
@@ -0,0 +1,664 @@
...
+      </tr>
+      <tr>
+        <td><a href="/why-coop-coep/#coop">COOP</a> violation</td>
+        <td>You've set a <code>Cross-Origin-Opener-Policy</code> on your site, but a cross-origin window is trying to interact directly with your document.</td>
```
Comment (agektmr) [inline src/site/content/en/blog/reporting-api/index.md:84]: Similar to above comment: I'd rephrase it to `You've set a <code>Cross-Origin-Opener-Policy</code> on the page, but a cross-origin window is trying to interact directly with the document.`

File: src/site/content/en/blog/reporting-api/index.md [hunk 35]
```diff
@@ -0,0 +1,664 @@
...
+      </tr>
+      <tr>
+        <td><a href="/why-coop-coep/#coep">COEP</a> violation</td>
+        <td>You've set a <code>Cross-Origin-Embedder-Policy</code> on your site, but a page on your site includes a cross-origin iframe that has not opted into being loaded by cross-origin documents.</td>
```
Comment (agektmr) [inline src/site/content/en/blog/reporting-api/index.md:88]: Similar to above comment: I'd rephrase it to `You've set a <code>Cross-Origin-Embedder-Policy</code> on a page, but it includes a cross-origin iframe that has not opted into being loaded by cross-origin documents.`

File: src/site/content/en/blog/reporting-api/index.md [hunk 36]
```diff
@@ -0,0 +1,664 @@
...
+      </tr>
+      <tr>
+        <td>Document Policy violation</td>
+        <td>Your site has a document policy that prevents usage of <code>document.write</code>, but a script on your page tries to use <code>document.write</code>.</td>
```
Comment (agektmr) [inline src/site/content/en/blog/reporting-api/index.md:92]: `Your site has a document policy that prevents usage of <code>document.write</code>, but a script on your page tries to use <code>document.write</code>.`
```suggestion
        <td>The page has a document policy that prevents usage of <code>document.write</code>, but a script tries to use <code>document.write</code>.</td>
```

File: src/site/content/en/blog/reporting-api/index.md [hunk 37]
```diff
@@ -0,0 +1,664 @@
...
+      </tr>
+      <tr>
+        <td>Deprecation warning</td>
+        <td>Your site is using an API that is deprecated, directly or via a top-level third-party script.
```
Comment (agektmr) [inline src/site/content/en/blog/reporting-api/index.md:96]:
```suggestion
        <td>The page is using an API that is deprecated, directly or via a top-level third-party script.
```

File: src/site/content/en/blog/reporting-api/index.md [hunk 38]
```diff
@@ -0,0 +1,664 @@
...
+        </td>
+      </tr>
+      <tr>
+        <td>Intervention</td>
+        <td>Your site is trying to do something that the browser decides not to honor, for security, performance or user experience reasons. Example in Chrome: your site uses <a href="https://developers.google.com/web/updates/2016/08/removing-document-write"><code>document.write</code> on slow networks</a> or uses <code>navigator.vibrate</code> in a cross-origin frame that the user hasn't interacted with yet.</td>
```
Comment (agektmr) [inline src/site/content/en/blog/reporting-api/index.md:101]: This one I'm not sure if it applies to the site or the page.

Comment (maudnals) [reply src/site/content/en/blog/reporting-api/index.md:101]: Thanks Eiji. I've included the suggestions. I'll check.

**Comment by maudnals** [comment]
@mihajlija The lint issue comes from a file that's not part of this PR. I couldn't find the tag that caused it.
